jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/ProjectCommand.java

```diff
@@ -296,25 +296,30 @@ public static String getModuleName(IJavaProject project) {
 
     public static boolean checkImportStatus() {
         IProject[] projects = ProjectUtils.getAllProjects();
+        boolean hasJavaProjectImported = false;
+        boolean hasError = false;
         for (IProject project : projects) {
             if (ProjectsManager.DEFAULT_PROJECT_NAME.equals(project.getName())) {
                 continue;
             }
+
+            if (ProjectUtils.isJavaProject(project)) {
+                hasJavaProjectImported = true;
+            }
+
             try {
-                List<IMarker> markers = ResourceUtils.getErrorMarkers(project);
-                if (markers != null) {
-                    boolean hasError = markers.stream().anyMatch(m -> {
-                        return m.getAttribute(IMarker.SEVERITY, 0) == IMarker.SEVERITY_ERROR;
-                    });
-                    if (hasError) {
-                        return true;
-                    }
+                int maxProblemSeverity = project.findMaxProblemSeverity(null, true, IResource.DEPTH_ONE);
+                if (maxProblemSeverity == IMarker.SEVERITY_ERROR) {
+                    hasError = true;
+                    break;
```

Comment:
Given a non-java project, could it have an error marker? If so, we will directly return `true` here. However, we should return `false` if there is another Java project exists.

Reply:
In line 322, it will only return true when no Java projects found in workspace.

But this reminds me that I should early return if a Java project is found in the for loop.